_posts/2020-09-21-permission-editor.md

```diff
@@ -0,0 +1,110 @@
+---
+
+title: "Permission editor in admin ui"
+author: [pusztaienike]
+image: "../img/posts/permission_editor.jpg"
+tags: [sensenet, permission]
+
+---
+
+"Nobody can hurt me without my permission. /Mahatma Gandhi/"
+
+---
+
+In the IT world you often handle secure data, which you don't want to share with the whole universe. But how can you hide those from the outsiders? What if you would like to share those only with your group? And another data with another group? 
+
+Of course there is a simple solution for that: permission system. Almost every software has permissions which handle what u can see & do.
+
+I think most of us use any social network. You may search for a person and you cannot see her/his photos, after you apply a friend request everything change.
+You see more than ever. Or what about your company? If you are not the boss I am sure that your company has datas which you do not have the sufficient permission.
+
+Permissions are everywhere. And of course sensenet has its own permission system as well. On the new admin-ui gui we did not have any surface to handle these permissions. Till now. 🕛 In the last couple of weeks we have tried to design and implement the simplest, easiest to use **permission editor**.
+
+## Customize it as you want 🔧🔨
+
+sensenet has a lot of built-in permission and we also keep some for your special needs:  Custom01, Custom02, etc. You can also group these permissions for the permission editor dialog as you want. The related settings file is available under Setup menu / Permission.settings.
+
+The default settings are the following:
+
+```{
+   "groups":[
+      {
+         "Read":["See", "Preview", "PreviewWithoutWatermark", "PreviewWithoutRedaction", "Open"]
+      },
+      {
+         "Write":[ "Save", "AddNew", "Delete"]
+      },
+      {
+         "Versioning":["OpenMinor", "Publish", "ForceCheckin", "Approve", "RecallOldVersion"]
+      },
+      {
+         "Advanced":["SeePermissions", "SetPermissions", "RunApplication"]
+      }
+   ]
+}
+```
+
+On the permission editor dialog you can see the groups and permissions based on these settings. 
+
+<p align="center">
+<img src="/img/posts/permission_groups.png">
+</p>
+
+## How to use it? 🎮
+
+Select a content from your repository, right-click on it and select 'Set permissions' from the dropdown. On the permission view you can check the entries which has inherited or set permissions. Selecting an entry makes the permission editor dialog open.
+
+On the dialog turn the switchers to on/off to enable/disable a permission.  On groups there are also switchers as well: you can easily enable/disable all of the permission under the selected group... and there is the king of kings: Full access 👑 (you can turn on/off all permissions)
+
+Some permissions are disabled because they are inherited and you cannot override them. The changes only apply on clicking Submit button so for your comfort there is also a reset button to restore the unsaved changes.
```

Comment:
```suggestion
When you open the dialog some permissions are disabled because they are inherited from a parent and you cannot override them. 
Changes only apply on clicking Submit button so for your comfort there is also a reset button to restore the unsaved changes.
```